### PR TITLE
Promote StatefulSetStartOrdinal to stable in 1.31

### PR DESCRIPTION
**Pull request by pwschuurman**
#### What type of PR is this?
/kind feature

#### What this PR does / why we need it:
This PR promotes the StatefulSetStartOrdinal feature to stable.

#### Does this PR introduce a user-facing change?
```release-note
Promote StatefulSetStartOrdinal to stable. This means `--feature-gates=StatefulSetStartOrdinal=true` are not needed on kube-apiserver and kube-controller-manager binaries and they'll be removed soon following policy at https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecation 
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
- KEP: https://github.com/kubernetes/enhancements/issues/3335
- Blog: https://github.com/kubernetes/website/pull/37418

### CODE REVIEW

File: api/discovery/apis.json [hunk 1]
```diff
@@ -281,6 +281,10 @@
         {
           "groupVersion": "flowcontrol.apiserver.k8s.io/v1beta3",
           "version": "v1beta3"
+        },
+        {
+          "groupVersion": "flowcontrol.apiserver.k8s.io/v1beta2",
+          "version": "v1beta2"
```
Comment (tengqm) [inline api/discovery/apis.json:287]: Is this change relevant to this PR?

Comment (pwschuurman) [reply api/discovery/apis.json:287]: I think this is just due to a stale openapi spec

Let me re-run `hack/update-openapi-spec.sh`

Comment (pwschuurman) [reply api/discovery/apis.json]: Removed these changes.

File: pkg/features/kube_features.go [hunk 2]
```diff
@@ -1167,7 +1168,7 @@ var defaultKubernetesFeatureGates = map[featuregate.Feature]featuregate.FeatureS
 
 	StatefulSetAutoDeletePVC: {Default: true, PreRelease: featuregate.Beta},
 
-	StatefulSetStartOrdinal: {Default: true, PreRelease: featuregate.Beta},
+	StatefulSetStartOrdinal: {Default: true, PreRelease: featuregate.GA, LockToDefault: true}, // GA in 1.31, remove in 1.32
```
Comment (soltysh) [inline pkg/features/kube_features.go:1171]: Nit, remove in 1.33, we usually do GA+2 for removal.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
for the sig-apps bits

File: pkg/features/kube_features.go (hunk 2 above)
Comment (pwschuurman) [reply pkg/features/kube_features.go]: Updated to 1.33 for removal.

**Review by hashim21223445 — APPROVED** [review]
